### ISSUE TITLE
Plugin crashes when child compiler has errors

### ISSUE THREAD

**Issue by meskill** [RoccoC/webpack-build-notifier#53]
Some other plugins can create [child compilers](https://webpack.js.org/api/compilation-object/#createchildcompiler) (for example [extract-css-chunks-webpack-plugin](https://github.com/faceyspacey/extract-css-chunks-webpack-plugin/blob/9daf720d11e671353d42d31fd05908ecae22af08/src/loader.js#L82)) which will lead to a crash with errors like this 
```
TypeError: Cannot read property 'module' of undefined
    at WebpackBuildNotifierPlugin.onCompilationDone
```
if that child compiler has errors or warnings

This is happening due to the code inside [webpack.Stats.hasErrors](https://github.com/webpack/webpack/blob/master/lib/Stats.js#L37) which checks errors also inside child compilers. 

To fix the issue we need to extract errors and warnings also from child compilers if there are no errors in the current one.

**Comment by RoccoC** [comment]
Thanks for the report.

Can you please provide a simple reproduce case for this scenario?

**Comment by meskill** [comment]
@RoccoC 
You can see reproduce for test cases here https://github.com/meskill/webpack-build-notifier/tree/bug-reproduce

**Comment by RoccoC** [comment]
@meskill , this has been fixed in `v2.1.0`.

Thanks for the bug report and reproduce case -- do let me know if you run into any issues with the latest release. 👍
